dfusion_rust_core/src/models/order.rs

```diff
@@ -91,11 +91,9 @@ impl From<Entity> for Order {
         let batch_information = entity
             .get("auctionId")
             .and(entity.get("slotIndex"))
-            .and_then(|_| {
-                Some(BatchInformation {
-                    slot: U256::from_entity(&entity, "auctionId"),
-                    slot_index: u16::from_entity(&entity, "slotIndex"),
-                })
+            .map(|_| BatchInformation {
+                slot: U256::from_entity(&entity, "auctionId"),
+                slot_index: u16::from_entity(&entity, "slotIndex"),
```

Comment:
Can you tell me what was failing the linter before this change?

Reply:
`and_then(|_| Some(x))` <=> `map(|_| x)`. The latter is simpler and suggested by clippy.

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#option_and_then_some